### ISSUE TITLE
Support for Tennis TV Replays

### ISSUE THREAD

**Issue by Wilann** [yt-dlp/yt-dlp#3192]
### Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running yt-dlp version **2022.03.08.1**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs [violate any copyrights](https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free) or contain any [DRM](https://en.wikipedia.org/wiki/Digital_rights_management) to the best of my knowledge
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and am willing to share it if required

### Region

Canada

### Example URLs

Playlist: https://www.tennistv.com/library/replays

### Description

I would like to download an archive of all tennis matches from Tennis TV. Perferable in format `/tournament/match`.mp4`, `/tournament/match2.mp4`, etc. Furthermore, it seems the tournament names for each match are written as some sort of "tag". Is there a way to extract that and download in my desired format? 

Currently, it seems the replays site is supported by yt-dlp. I would really appreciate it if support could be added! Thank you for your time. 

### Verbose log

```shell
yt-dlp https://www.tennistv.com/library/replays --verbose
[debug] Command-line config: ['https://www.tennistv.com/library/replays', '--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.03.08.2 [e491d06d3]
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.15.15-76051515-generic-x86_64-with-glibc2.31
[debug] exe versions: ffmpeg 4.2.4, ffprobe 4.2.4
[debug] Optional libraries: brotli, Cryptodome, secretstorage, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [generic] Extracting URL: https://www.tennistv.com/library/replays
[generic] replays: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] replays: Downloading webpage
[generic] replays: Extracting information
[debug] Looking for video embeds
ERROR: Unsupported URL: https://www.tennistv.com/library/replays
Traceback (most recent call last):
  File "/home/wilsonchan/anaconda3/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1389, in wrapper
    return func(self, *args, **kwargs)
  File "/home/wilsonchan/anaconda3/lib/python3.9/site-packages/yt_dlp/YoutubeDL.py", line 1459, in __extract_info
    ie_result = ie.extract(url)
  File "/home/wilsonchan/anaconda3/lib/python3.9/site-packages/yt_dlp/extractor/common.py", line 617, in extract
    ie_result = self._real_extract(url)
  File "/home/wilsonchan/anaconda3/lib/python3.9/site-packages/yt_dlp/extractor/generic.py", line 3998, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.tennistv.com/library/replays
```

**Comment by Bricio** [comment]
I am still deciding if i am going to code this.

but if anyone want to do it. go for it.
basic algorithm for what i saw in developer tools.
```
download webpage url
get all sections inside div that have class=body-content

entries=[]
for section in sections
    if data-widget=="video-listing/video-grid"
         i = 1
         while i <=data_num_pages # site uses 20 items steps, api accepts max 1000 items  
            json_url = https://api.tennistv.com/content/atpmedia/video/EN/?page=<i-1>&pageSize=<data-page-size>&tagNames=<data-tags>&references=<data-references>
           download json
           for object in json.contents:
                entries.append(f'https://www.tennistv.com/videos/{object.id}/{object.titleUrlSegment}')
           i += 1

     elif data-widget=="video-listing/video-rail-playlist"
          json_url = https://api.tennistv.com/content/atpmedia/playlist/en/<data-playlist-id>?detail=DETAILED&pageSize=<data-page-size>&fullObjectResponse=true
          download json
          for object in json.items
              entries.append(f'https://www.tennistv.com/videos/{object.response.id}/{object.response.titleUrlSegment}')

     else
           something not know....

return self.playlist_result(entries,......)
```

btw... all replays is a 10800+ videos playlist.
`"pageInfo":{"page":0,"numPages":541,"pageSize":20,"numEntries":10801}`

good luck downloading that....

**Comment by pukkandan** [comment]
Does individual replays already download?

> Furthermore, it seems the tournament names for each match are written as some sort of "tag". Is there a way to extract that and download in my desired format?

I don't understand. Are you asking about individual matches here? If so, please open a new issue with relevant verbose log. Note that since the site requires login, I cannot test anything, and so cannot help unless you can provide sufficient information

**Comment by Wilann** [comment]
> Does individual replays already download?

When trying to download an individual replay, I get an error:
```
[debug] Command-line config: ['https://www.tennistv.com/videos/2553362/miami-2022-r3-tsitsipas-de-minaur', '--username', 'PRIVATE', '--password', 'PRIVATE', '--verbose']
[debug] User config "/Users/wilsonchan/.config/yt-dlp/config": ['--add-metadata']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, err UTF-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef]
[debug] Python version 3.7.4 (CPython 64bit) - Darwin-21.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[TennisTV] Logging in
ERROR: [TennisTV] 2553362: Login failed (wrong password?): HTTP Error 403: Forbidden (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U (caused by <HTTPError 403: 'Forbidden'>); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/site-packages/yt_dlp/extractor/common.py", line 730, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/site-packages/yt_dlp/YoutubeDL.py", line 3558, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/urllib/request.py", line 641, in http_response
    'http', request, response, code, msg, hdrs)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/Users/wilsonchan/opt/anaconda3/lib/python3.7/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
```

------------

>> Furthermore, it seems the tournament names for each match are written as some sort of "tag". Is there a way to extract that and download in my desired format?

Apologies for the confusion - below I've attached an image where I've underlined (in red) the tournament, year, and round number for each match, which I would like to use to format the output filename as something like "upload_date - tournament round - match_name" (for example, "2022XXXX - Miami R3 - Stefanos Tsitsipas vs Alex de Minaur")

<img width="1303" alt="Screen Shot 2022-03-29 at 9 43 37 AM" src="https://user-images.githubusercontent.com/40744757/160625773-6987beed-ddc7-4330-afa6-c192e63a87db.png">

----------

> I don't understand. Are you asking about individual matches here? If so, please open a new issue with relevant verbose log. 

I would like to download all videos (individual matches) with a single command, given the replays URL: https://www.tennistv.com/library/replays 

----------

> Note that since the site requires login, I cannot test anything, and so cannot help unless you can provide sufficient information

I would be glad to provide my login info - can I give it to you via email or via some other private method?

**Comment by zenerdi0de** [comment]
> Does individual replays already download?

Individual replays/highlights should be supported once https://github.com/yt-dlp/yt-dlp/pull/2324 is merged.

**Comment by Wilann** [comment]
Hi @pukkandan, just wondering if there's any update on this. If an account is needed, how can I provide it?

**Comment by pukkandan** [comment]
You can see https://github.com/yt-dlp/yt-dlp/pull/2324 is pending fixes from my review. Don't ping me unnecessarily

**Comment by gray-morley** [comment]
Just tried downloading a replay and it gave an error, something about a Ksession, does that mean support for tennisTV is dead?
